maat-court-data-api/src/main/java/gov/uk/courtdata/reporder/validator/CCOutComeValidationProcessor.java

```diff
@@ -0,0 +1,33 @@
+package gov.uk.courtdata.reporder.validator;
+
+import com.amazonaws.xray.spring.aop.XRayEnabled;
+import gov.uk.courtdata.model.RepOrderCCOutcome;
+import gov.uk.courtdata.validator.MaatIdValidator;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@XRayEnabled
+public class CCOutComeValidationProcessor {
+
+    private final MaatIdValidator maatIdValidator;
+    private final CreateCCOutcomeValidator createCCOutComeValidator;
+
+    public Optional<Void> validate(RepOrderCCOutcome repOrderCCOutCome) {
+
+        if (null == repOrderCCOutCome.getId() || repOrderCCOutCome.getId() == 0) {
+            createCCOutComeValidator.validate(repOrderCCOutCome);
+        }
+        return maatIdValidator.validate(repOrderCCOutCome.getRepId());
+    }
+
+    public Optional<Void> validate(Integer repId) {
+        return maatIdValidator.validate(repId);
+    }
```

Comment:
I think I'd rather us just use the maatIdValidator rather than proxying the call, but it's not a massive problem.

Reply:
Good to have all the validation in once place hence not moving maatIdValidator to controller